containers/Itunes/index.tsx

```diff
@@ -19,19 +19,24 @@ const Container = styled.div`
 
 export const TracksContainer = () => {
   const [searchTerm, setSearchTerm] = useState<string>("");
-  const { data } = useFetchTracksQuery(searchTerm, { skip: !searchTerm });
+  const { data, error, isLoading, isFetching, isSuccess } = useFetchTracksQuery(searchTerm, {
+    skip: !searchTerm,
+  });
 
   const handleOnChange = trackName => setSearchTerm(trackName);
 
   const debouncedHandleOnChange = debounce(handleOnChange, 200);
 
-  // console.log("data", data);
-
   return (
-    <Container>
-      <Header />
-      <SearchBox debouncedHandleOnChange={debouncedHandleOnChange} />
-    </Container>
+    <>
+      <Container>
+        <Header />
+        <SearchBox debouncedHandleOnChange={debouncedHandleOnChange} />
+        {error && message.error("There's an error while fetching the tracks...")}
+        {isFetching && <Spin />}
```

Comment:
This will work. But can also use the pros that Antd Spin provides to say if it should load or not load.
I think passing isFetching to spinning prop in Spin will do the trick

Reply:
Thanks for the update. Updated the change accordingly.